tiledb/bioimg/converters/base.py

```diff
@@ -536,7 +544,12 @@ def to_tiledb(
                 original_metadata = reader.original_metadata
             else:
                 if ome_xml := reader.original_metadata.get("ome_metadata"):
-                    pruned_metadata = remove_ome_image_metadata(ome_xml)
+                    if isinstance(exclude_metadata, bool):
+                        pruned_metadata = remove_ome_image_metadata(ome_xml)
+                    elif callable(exclude_metadata):
+                        pruned_metadata = exclude_metadata(ome_xml)
```

Comment:
@ktsitsi callbacks all tricky. It's almost guaranteed that accidents will happen. I think it would be good to have some extra precise error handling here. Something like this:

```suggestion
                        try:
                            pruned_metadata = exclude_metadata(ome_xml)
                        except Exception as exc:
                            raise MetadataCallbackError("OME XML callback function failed") from exc
```

Reply:
Thank you @sgillies. nice catch! I added the changes.